src/components/task/existing-task/task-card/index.jsx

```diff
@@ -29,31 +29,37 @@ function TaskCard({
 
   const taskHeaderClasses = classnames({ "text-line-through": isCompleted });
 
-  return (
-    <div className={styles.card}>
-      {isTextAreaVisible ? (
-        <TextArea numOfRows={TASK_TEXTAREA_NUM_OF_ROW} />
-      ) : (
-        <h3 className={taskHeaderClasses}>{title}</h3>
-      )}
-
-      <p className={styles.date}>Created At: {formatDate(createdAt)}</p>
-
-      <div className="flex justify-between">
-        <ButtonContainer
-          onDoneButtonClick={() => dispatch(markAsDone(taskId))}
-          onEditButtonClick={() => setIsTextAreaVisible(true)}
-          onDeleteButtonClick={() => dispatch(deleteTask(taskId))}
-          isTaskCompleted={isCompleted}
+  if (isTextAreaVisible) {
+    return (
+      <div className={styles.card}>
+        <EditTaskForm
+          taskId={taskId}
+          existingTitle={title}
+          onDelete={() => setIsTextAreaVisible(false)}
+          onTaskEdit={() => setIsTextAreaVisible(false)}
         />
-        {isCompleted && (
-          <Button>
-            Completed in {calculateDateDifference(completedAt, createdAt)}
-          </Button>
-        )}
       </div>
-    </div>
-  );
+    );
+  } else
+    return (
+      <div className={styles.card}>
+        <h3 className={taskHeaderClasses}>{title}</h3>
+        <p className={styles.date}>Created At: {formatDate(createdAt)}</p>
+        <div className="flex justify-between">
+          <ButtonContainer
+            onDone={() => dispatch(markAsDone(taskId))}
+            onEdit={() => setIsTextAreaVisible(true)}
+            onDelete={() => dispatch(deleteTask(taskId))}
+            isTaskCompleted={isCompleted}
+          />
+          {isCompleted && (
+            <Button>
+              Completed in {calculateDateDifference(completedAt, createdAt)}
+            </Button>
+          )}
+        </div>
+      </div>
+    );
```

Comment:
use consistent brace style.

```suggestion
  } else {
    return (
        <div className={styles.card}>
          <h3 className={taskHeaderClasses}>{title}</h3>
          <p className={styles.date}>Created At: {formatDate(createdAt)}</p>
          <div className="flex justify-between">
            <ButtonContainer
                onDone={() => dispatch(markAsDone(taskId))}
                onEdit={() => setIsTextAreaVisible(true)}
                onDelete={() => dispatch(deleteTask(taskId))}
                isTaskCompleted={isCompleted}
            />
            {isCompleted && (
                <Button>
                  Completed in {calculateDateDifference(completedAt, createdAt)}
                </Button>
            )}
          </div>
        </div>
    );
  }
```